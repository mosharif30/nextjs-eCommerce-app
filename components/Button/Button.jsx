import React from 'react'

/** @jsxImportSource @emotion/react */
import { css, useTheme } from '@emotion/react'

function Button({ type = 'button', children, secondary, sec, textWhite }) {
  const theme = useTheme()
  return (
    <button
      type={type}
      css={css`
        width: 100%;
        background-color: ${secondary
          ? theme.colors.secondary
          : sec
          ? theme.colors.sec
          : theme.colors.primary};
        color: ${textWhite ? theme.colors.white : theme.colors.text};
        font-size: ${theme.typography.paragraph.fontSize};
        text-decoration: none;
        padding: ${theme.spaces[4]};
        border: ${theme.borderRadius[1]};
        display: inline-block;
        position: relative;
        cursor: pointer;
        transition-duration: 0.4s;
        &:hover {
          box-shadow: ${theme.boxShadow.light};
          background-color: ${theme.colors.secondary};
        }
      `}
    >
      {children}
    </button>
  )
}

export default Button
