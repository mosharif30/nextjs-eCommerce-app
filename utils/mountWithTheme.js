import { ThemeProvider } from '@emotion/react'
import theme from '../configs/theme'
import renderer from 'react-test-renderer'

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  )
}
export default mountWithTheme
