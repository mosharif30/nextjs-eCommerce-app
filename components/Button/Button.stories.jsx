import React from 'react'

/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react'
import { useTheme } from '@emotion/react'
import Button from './Button'

export default { title: 'Button' }
export const primary = () => <Button> کلیک کنید</Button>
export const secondary = () => <Button secondary>قرمز</Button>
