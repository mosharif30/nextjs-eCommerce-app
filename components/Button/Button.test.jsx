import mountWithTheme from '../../utils/mountWithTheme'
import Button from './Button'
describe('button', () => {
  test('render correctly', () => {
    const component = mountWithTheme(<Button>sample Button</Button>)
    const tree = component
    expect(tree).toMatchSnapshot()
  })
  test('render correctly secondary', () => {
    const component = mountWithTheme(
      <Button secondary>sample Button secondary</Button>
    )
    const tree = component
    expect(tree).toMatchSnapshot()
  })
  test('render correctly loading', () => {
    const component = mountWithTheme(
      <Button loading>sample Button loading</Button>
    )
    const tree = component
    expect(tree).toMatchSnapshot()
  })
  test('render correctly danger', () => {
    const component = mountWithTheme(
      <Button danger>sample Button danger</Button>
    )
    const tree = component
    expect(tree).toMatchSnapshot()
  })
})
