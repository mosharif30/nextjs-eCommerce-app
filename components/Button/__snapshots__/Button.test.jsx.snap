// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1o20tv1-Button{width:100%;background-color:#ecf0f1;color:#2c3e50;font-size:1em;-webkit-text-decoration:none;text-decoration:none;padding:16px;border:2px;display:inline-block;position:relative;cursor:pointer;transition-duration:0.4s;}.css-1o20tv1-Button:hover{box-shadow:0 4px 16px rgba(0,1,31,0.2);background-color:#16a085;}",
      }
    }
    data-emotion="css 1o20tv1-Button"
  />,
  <button
    className="css-1o20tv1-Button"
    type="button"
  >
    sample Button
  </button>,
]
`;

exports[`button render correctly danger 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1ge2ywf-Button{width:100%;background-color:#e74c3c;color:#2c3e50;font-size:1em;-webkit-text-decoration:none;text-decoration:none;padding:16px;border:2px;display:inline-block;position:relative;cursor:pointer;transition-duration:0.4s;}.css-1ge2ywf-Button:hover{box-shadow:0 4px 16px rgba(0,1,31,0.2);background-color:#c0392b;}",
      }
    }
    data-emotion="css 1ge2ywf-Button"
  />,
  <button
    className="css-1ge2ywf-Button"
    type="button"
  >
    sample Button danger
  </button>,
]
`;

exports[`button render correctly loading 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1o20tv1-Button{width:100%;background-color:#ecf0f1;color:#2c3e50;font-size:1em;-webkit-text-decoration:none;text-decoration:none;padding:16px;border:2px;display:inline-block;position:relative;cursor:pointer;transition-duration:0.4s;}.css-1o20tv1-Button:hover{box-shadow:0 4px 16px rgba(0,1,31,0.2);background-color:#16a085;}",
      }
    }
    data-emotion="css 1o20tv1-Button"
  />,
  <button
    className="css-1o20tv1-Button"
    type="button"
  >
    صبر کنید
  </button>,
]
`;

exports[`button render correctly secondary 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1d5avbz-Button{width:100%;background-color:#16a085;color:#2c3e50;font-size:1em;-webkit-text-decoration:none;text-decoration:none;padding:16px;border:2px;display:inline-block;position:relative;cursor:pointer;transition-duration:0.4s;}.css-1d5avbz-Button:hover{box-shadow:0 4px 16px rgba(0,1,31,0.2);background-color:#16a085;}",
      }
    }
    data-emotion="css 1d5avbz-Button"
  />,
  <button
    className="css-1d5avbz-Button"
    type="button"
  >
    sample Button secondary
  </button>,
]
`;
